### PR TITLE
(MODULES-7067) Only initialise constant when not defined

### PR DESCRIPTION
**Pull request by btoonk**
https://tickets.puppetlabs.com/browse/MODULES-7067

### CODE REVIEW

File: lib/puppet/provider/exec/powershell.rb [hunk 1]
```diff
@@ -27,28 +27,30 @@
...
 
-  - PowerShell v2 with .NET Framework 2.0
+    - PowerShell v2 with .NET Framework 2.0
 
-    PowerShell v2 works with both .NET Framework 2.0 and .NET Framework 3.5.
-    To be able to use the enhancements, we require .NET Framework 3.5.
-    Typically you will only see this on a base Windows Server 2008 (and R2)
-    install.
+      PowerShell v2 works with both .NET Framework 2.0 and .NET Framework 3.5.
+      To be able to use the enhancements, we require at least .NET Framework 3.5.
```
Comment (glennsarti) [inline lib/puppet/provider/exec/powershell.rb]: `at least` was removed in commit https://github.com/puppetlabs/puppetlabs-powershell/commit/18628d6d4dc690dfe52430f53ebbfc2dbdc02e95. This appears to be revertting that.

File: lib/puppet/provider/exec/powershell.rb [hunk 2]
```diff
@@ -27,28 +27,30 @@
         }
   EOT
 
-  POWERSHELL_MODULE_UPGRADE_MSG = <<-UPGRADE
-  Currently, the PowerShell module has reduced v1 functionality on this agent
-  due to one or more of the following conditions:
+  unless defined?(POWERSHELL_MODULE_UPGRADE_MSG)
+    POWERSHELL_MODULE_UPGRADE_MSG = <<-UPGRADE
```
Comment (Iristyle) [inline lib/puppet/provider/exec/powershell.rb]: You might be able to use `||=` instead of `unless defined?`

**Comment by Iristyle** [comment]
For sure this should fix a redefinition.

However, this may be symptomatic of a double loading problem, where the source files are required more than once. One way to determine this would be to put a debugging statement that emits the callstack like:

```ruby
puts "Constant defined #{caller.join("\n")}"
```

That would then allow us to figure out why this code is being loaded multiple times, when it should only be loaded once.

**Comment by btoonk** [comment]
@Iristyle We only see this when using puppet device with multiple devices in a singe device configuration file.

The debug statement output:
Constant defined /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/classgen.rb:136:in `class_eval'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/classgen.rb:136:in `genthing'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/classgen.rb:36:in `genclass'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/type.rb:1860:in `provide'
/opt/puppetlabs/puppet/cache/devices/device-x/lib/puppet/provider/exec/powershell.rb:5:in `<top (required)>'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/autoload.rb:68:in `load'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/autoload.rb:68:in `load_file'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/autoload.rb:88:in `block in reload_changed'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/autoload.rb:88:in `each'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/autoload.rb:88:in `reload_changed'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/configurer/plugin_handler.rb:21:in `download_plugins'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/configurer.rb:444:in `download_plugins'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/configurer.rb:117:in `get_facts'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/configurer.rb:306:in `run_internal'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/configurer.rb:221:in `block in run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/context.rb:65:in `override'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet.rb:306:in `override'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/configurer.rb:195:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/device.rb:209:in `block in main'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/device.rb:183:in `each'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/device.rb:183:in `collect'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/device.rb:183:in `main'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:383:in `run_command'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:375:in `block in run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util.rb:662:in `exit_on_fail'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:375:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/command_line.rb:132:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/command_line.rb:72:in `execute'
/usr/bin/puppet:5:in `<main>'

File: lib/puppet/provider/exec/powershell.rb (hunk 2 above)
Comment (btoonk) [reply lib/puppet/provider/exec/powershell.rb]: @Iristyle I was unaware of the existence of this operator. I modified the commit to use this operator. Thanks for pointing this out.

**Pull request merged**